src/views/forms/AccountManagementFormsNew/WithdrawForm/useValidation.tsx

```diff
@@ -0,0 +1,203 @@
+import { useMemo } from 'react';
+
+import { exchange } from '@dydxprotocol/v4-abacus';
+import { Asset } from '@skip-go/client';
+import BigNumber from 'bignumber.js';
+
+import { AlertType } from '@/constants/alerts';
+import { STRING_KEYS } from '@/constants/localization';
+import {
+  MAX_CCTP_TRANSFER_AMOUNT,
+  MIN_CCTP_TRANSFER_AMOUNT,
+  TOKEN_DECIMALS,
+} from '@/constants/numbers';
+
+import { useLocaleSeparators } from '@/hooks/useLocaleSeparators';
+import { useRestrictions } from '@/hooks/useRestrictions';
+import { useStringGetter } from '@/hooks/useStringGetter';
+import { useTokenConfigs } from '@/hooks/useTokenConfigs';
+import { useWithdrawalInfo } from '@/hooks/useWithdrawalInfo';
+
+import { formatNumberOutput, OutputType } from '@/components/Output';
+import { Tag } from '@/components/Tag';
+
+import { useAppSelector } from '@/state/appTypes';
+import { getSelectedLocale } from '@/state/localizationSelectors';
+
+import { MustBigNumber } from '@/lib/numbers';
+
+type UseValidationProps = {
+  isCctp: boolean;
+  debouncedAmountBN: BigNumber;
+  freeCollateralBN: BigNumber;
+  isValidDestinationAddress: boolean;
+  error?: string;
+  toAddress?: string;
+  toChainId?: string;
+  toToken?: Asset;
+};
+
+export const useValidation = ({
+  error,
+  isCctp,
+  debouncedAmountBN,
+  toAddress,
+  isValidDestinationAddress,
+  toChainId,
+  toToken,
+  freeCollateralBN,
+}: UseValidationProps) => {
+  const stringGetter = useStringGetter();
+  const { sanctionedAddresses } = useRestrictions();
+  const { decimal: decimalSeparator, group: groupSeparator } = useLocaleSeparators();
+  const selectedLocale = useAppSelector(getSelectedLocale);
+  const { usdcLabel } = useTokenConfigs();
+  const { usdcWithdrawalCapacity } = useWithdrawalInfo({ transferType: 'withdrawal' });
+
+  const { alertType, errorMessage } = useMemo(() => {
```

Comment:
return here instead of on line 199 - feels unnecessary to destructure and then restructure